### PR TITLE
Fixes vendor showing weird glass icons

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Fixes vendor showing weird glass icons
## Why It's Good For The Game
bug fix good
## Changelog
:cl:
fix: vendor no longer shows glass icons in a specific condition
/:cl:

### CODE REVIEW

File: code/modules/asset_cache/assets/vending.dm [hunk 1]
```diff
@@ -2,7 +2,16 @@
 	name = "vending"
 
 /datum/asset/spritesheet/vending/create_spritesheets()
-	for (var/k in GLOB.vending_products)
+	// initialising the list of items we need
+	var/target_items = list()
+	for(var/obj/machinery/vending/V as() in typesof(/obj/machinery/vending))
```
Comment (Mothblocks) [inline code/modules/asset_cache/assets/vending.dm:7]: Real variable names

`as anything`, not `as()`

File: code/modules/asset_cache/assets/vending.dm [hunk 2]
```diff
@@ -2,7 +2,16 @@
...
+		V = new V() // It seems `initial(list var)` has nothing. need to make a type.
+		for(var/O in list(V.products, V.premium, V.contraband))
```
Comment (Mothblocks) [inline code/modules/asset_cache/assets/vending.dm:9]: Real variable name

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I don't understand how this fixes anything, nothing here is `initial`, `vending_products` is built from real vending machines.

**Comment by EvilDragonfiend** [comment]
> I don't understand how this fixes anything, nothing here is `initial`, `vending_products` is built from real vending machines.

`vending_products` glob is established with only existing vending machines when the server is booted. that means if there's no real vending machine, `vending_products` has nothing and it fails to make the image asset.
For example, if you run a runtime station, only the vending machines there have the icons, and additional vending machines you place will not have the proper icons.

**Comment by Mothblocks** [comment]
I see

Also this is going to poop on init times a bit but we're going to be killing this entire thing with 515 anyway so whatevs

**Pull request merged**